File: tyk/middleware/add-channel.js
// Add Channel Header Middleware
// Equivalent to Apigee: Add-Channel Policy

var addChannelHeader = new TykJS.TykMiddleware.NewMiddleware({});

addChannelHeader.NewProcessRequest(function(request, session, config) {

    // Default channel
    var channel = "Web";

    // Read from API key metadata
    if (session && session.meta_data && session.meta_data.channel) {
        channel = session.meta_data.channel;
    }

    // Inject into headers
    request.SetHeaders["X-CHANNEL"] = channel;

    // Apigee-compat prefix header
    request.SetHeaders["X-Forwarded-Prefix"] = "/dummy-proxy/v1";

    log("Added X-CHANNEL header: " + channel);

    return addChannelHeader.ReturnData(request, session.meta_data);
});

log("Add-channel middleware loaded");

File: tyk-config/middleware/dummy/add_channel/add_channel.go
package main

import (
	"net/http"
)

// AddChannelHeader is the pre-request Go plugin equivalent of add-channel.js.
// It injects X-CHANNEL and X-Forwarded-Prefix headers before the request
// is forwarded to the upstream backend.
// Tyk calls this function by name as defined in the API definition.
func AddChannelHeader(rw http.ResponseWriter, r *http.Request) {
	channel := "Web" // default channel

	// Override from X-Session-Channel if set upstream
	if sessionChannel := r.Header.Get("X-Session-Channel"); sessionChannel != "" {
		channel = sessionChannel
	}

	r.Header.Set("X-CHANNEL", channel)
	r.Header.Set("X-Forwarded-Prefix", "/dummy-proxy/v1")
}

func main() {}
